lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Runtime.h"
+#include <time.h>
+#include <sys/time.h>
+#include "ChakraPlatform.h"
+
+namespace Js
+{
+    #define updatePeriod 1000
+
+    const WCHAR *Utility::GetStandardName(size_t *nameLength) {
+        UpdateTimeZoneInfo();
+        *nameLength = standardZoneNameLength;
+        return standardZoneName;
+    }
+
+    const WCHAR *Utility::GetDaylightName(size_t *nameLength) {
+        // We have an abbreviated standard or daylight name
+        // based on the date and time zone we are in.
+        return GetStandardName(nameLength);
+    }
+
+    void Utility::UpdateTimeZoneInfo()
+    {
+        uint32 tickCount = GetTickCount();
+        if (tickCount - lastTimeZoneUpdateTickCount > updatePeriod)
+        {
+            time_t gtime = time(NULL);
+            struct tm ltime;
+
+            tzset();
+            localtime_r(&gtime, &ltime);
+
+            standardZoneNameLength = strlen(ltime.tm_zone);
+
+        #if defined(WCHAR_IS_CHAR16_T)
+            for(int i = 0; i < standardZoneNameLength; i++) {
+                standardZoneName[i] = (char16_t)ltime.tm_zone[i];
+            }
+        #elif defined(WCHAR_IS_WCHAR_T)
+            mbstowcs( (wchar_t*) standardZoneName, ltime.tm_zone,
+                      sizeof(standardZoneName) / sizeof(WCHAR));
+        #else
+        #error "WCHAR should be either wchar_t or char16_t"
+        #endif
+
+            standardZoneName[standardZoneNameLength] = (WCHAR)0;
+            lastTimeZoneUpdateTickCount = tickCount;
+        }
+    }
+
+    // DateTimeHelper ******
+    #define TM_TO_SYSTIME(SYS, TM)           \
+            SYS->wYear = TM->tm_year + 1900; \
+            SYS->wMonth = TM->tm_mon + 1;    \
+            SYS->wDayOfWeek = TM->tm_wday;   \
+            SYS->wDay = TM->tm_mday;         \
+            SYS->wHour = TM->tm_hour;        \
+            SYS->wMinute = TM->tm_min;       \
+            SYS->wSecond = TM->tm_sec
+
+    #define SYSTIME_TO_TM(SYS, TM)           \
+            TM->tm_year = SYS->wYear - 1900; \
+            TM->tm_mon = SYS->wMonth - 1;    \
+            TM->tm_wday = SYS->wDayOfWeek;   \
+            TM->tm_mday = SYS->wDay;         \
+            TM->tm_hour = SYS->wHour;        \
+            TM->tm_min = SYS->wMinute;       \
+            TM->tm_sec = SYS->wSecond
+
+    const bool SysLocalToUtc(SYSTEMTIME *local, SYSTEMTIME *utc)
+    {
+        struct tm local_tm;
+        SYSTIME_TO_TM(local, (&local_tm));
+
+        // tm doesn't have milliseconds
+        // uint32_t msec = utc->wMilliseconds;
+        // simply keep it as is
+
+        tzset();
+        time_t utime = timegm(&local_tm);
```

Comment:
Curious, why can't you just do the following:
```local_tm.tm_isdst = daylight; // set by tzset
time_t localTime = mktime(&local_tm);
tm* utcTm = gmtime(&localTime);
```

At any rate, the code here is non-trivial enough that it's worth writing a detailed comment here. I suspect the convoluted approach here is to avoid figuring out if dst is in effect right now and instead just pretending like the local tm was UTC, calculating the calendar time for that, then using mktime to just figure out the offset (taking into account whether dst is in effect), and adding that difference to the utc calendar time. I think my snippet above does basically the same thing. Also don't follow why you copy local_tm to local_bck here since it seems like local_tm is not used after the copy anyway